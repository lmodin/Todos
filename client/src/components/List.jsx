import React from 'react';
import ListItem from './ListItem.jsx'

const List = (props) => {
  //console.log('list is loaded: ',props.isLoaded)
  if (!props.isLoaded) {
    return (
      <div>
        Loading your To Do items
      </div>
    )
  } else if (props.items.length === 0) {
    return (
      <div>
        Congratulations! You have no outstanding to dos.
      </div>
    )
  } else {
    return (
      <div>
        Click on an item to remove it
        <ul className="list" >
          {props.items.map((item,index) => (
            <ListItem item={item} key={item+index}removeToDo={props.removeToDo} />
          ))}
        </ul>
      </div>
    )
  }
}

export default List