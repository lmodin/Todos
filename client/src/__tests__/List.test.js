import React from 'react';
//import { cleanup, fireEvent, render} from '@testing-library/react';
import List from '../components/List.jsx';
import renderer from 'react-test-renderer';

//afterEach(cleanup);

//test list of todos
const testToDos = [
  'Do the laundry',
  'Make lunch',
  'Water the garden'
]


//snapshot testing
it('renders correctly', () => {
  const list = renderer
    .create(<List isLoaded={true} items={testToDos}></List>)
    .toJSON();
  expect(list).toMatchSnapshot();
})