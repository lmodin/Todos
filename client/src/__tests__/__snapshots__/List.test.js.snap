// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  Click on an item to remove it
  <ul
    className="list"
  >
    <li
      onClick={[Function]}
    >
      Do the laundry
    </li>
    <li
      onClick={[Function]}
    >
      Make lunch
    </li>
    <li
      onClick={[Function]}
    >
      Water the garden
    </li>
  </ul>
</div>
`;
